cpp/src/search/search.cu

```diff
@@ -248,83 +237,115 @@ bool contains_scalar_dispatch::operator()<cudf::dictionary32>(column_view const&
+      thrust::uninitialized_fill(rmm::exec_policy(stream), out_begin, out_begin + n_rows, false);
       return result;
     }
 
-    auto hash_set        = cudf::detail::unordered_multiset<Element>::create(needles, stream);
-    auto device_hash_set = hash_set.to_device();
+    auto hash_set          = cudf::detail::unordered_multiset<Type>::create(needles, stream);
+    auto const hash_set_dv = hash_set.to_device();
 
-    auto d_haystack_ptr = column_device_view::create(haystack, stream);
-    auto d_haystack     = *d_haystack_ptr;
+    auto const haystack_cdv_ptr = column_device_view::create(haystack, stream);
+    auto const haystack_cdv     = *haystack_cdv_ptr;
+    auto const begin            = thrust::make_counting_iterator<size_type>(0);
+    auto const end              = begin + haystack.size();
 
     if (haystack.has_nulls()) {
       thrust::transform(rmm::exec_policy(stream),
-                        thrust::make_counting_iterator<size_type>(0),
-                        thrust::make_counting_iterator<size_type>(haystack.size()),
-                        result_view.begin<bool>(),
-                        [device_hash_set, d_haystack] __device__(size_t index) {
-                          return d_haystack.is_null_nocheck(index) ||
-                                 device_hash_set.contains(d_haystack.element<Element>(index));
+                        begin,
+                        end,
+                        out_begin,
+                        [hash_set_dv, haystack_cdv] __device__(size_t idx) {
+                          return haystack_cdv.is_null_nocheck(idx) ||
+                                 hash_set_dv.contains(haystack_cdv.template element<Type>(idx));
                         });
     } else {
       thrust::transform(rmm::exec_policy(stream),
-                        thrust::make_counting_iterator<size_type>(0),
-                        thrust::make_counting_iterator<size_type>(haystack.size()),
-                        result_view.begin<bool>(),
-                        [device_hash_set, d_haystack] __device__(size_t index) {
-                          return device_hash_set.contains(d_haystack.element<Element>(index));
+                        begin,
+                        end,
+                        out_begin,
+                        [hash_set_dv, haystack_cdv] __device__(size_t index) {
+                          return hash_set_dv.contains(haystack_cdv.template element<Type>(index));
                         });
     }
 
     return result;
   }
-};
 
-template <>
-std::unique_ptr<column> multi_contains_dispatch::operator()<list_view>(
-  column_view const&, column_view const&, rmm::cuda_stream_view, rmm::mr::device_memory_resource*)
-{
-  CUDF_FAIL("list_view type not supported");
-}
+  template <typename Type>
+  std::enable_if_t<is_nested<Type>(), std::unique_ptr<column>> operator()(
+    column_view const& values /* => haystack */,
+    column_view const& input /* => needles */,
```

Comment:
Can we just call these haystack/needles instead of changing their name with a comment telling us what they are?

Reply:
I want (and I'm going to have) another PR that completely and consistently changes these into `search_space` and `search_values`. The new names are more accurate and descriptive.

Reply:
I like haystack/needles, they should just be used correctly as you pointed out how they are definitely not currently!